src/wolff_producer.erl

```diff
@@ -630,24 +631,33 @@ collect_send_calls(Calls, Count, Size, Limit) ->
 enqueue_calls(Calls, #{replayq := Q,
                        pending_acks := PendingAcks0,
                        call_id_base := CallIdBase,
-                       partition := Partition
+                       partition := Partition,
+                       config := Config0
                       } = St0) ->
-  {QueueItems, PendingAcks} =
+  {QueueItems, PendingAcks, CallByteSize} =
     lists:foldl(
-      fun(?SEND_REQ(_From, Batch, AckFun), {Items, PendingAcksIn}) ->
+      fun(?SEND_REQ(_From, Batch, AckFun), {Items, PendingAcksIn, Size}) ->
           CallId = make_call_id(CallIdBase),
           %% keep callback funs in memory, do not seralize it into queue because
           %% saving anonymous function to disk may easily lead to badfun exception
           %% in case of restart on newer version beam.
           PendingAcksOut = PendingAcksIn#{CallId => ?ACK_CB(AckFun, Partition)},
           NewItems = [make_queue_item(CallId, Batch) | Items],
-          {NewItems, PendingAcksOut}
-      end, {[], PendingAcks0}, Calls),
+          {NewItems, PendingAcksOut, Size + batch_bytes(Batch)}
+      end, {[], PendingAcks0, 0}, Calls),
    NewQ = replayq:append(Q, lists:reverse(QueueItems)),
    lists:foreach(fun maybe_reply_queued/1, Calls),
+   Overflow = case maps:get(drop_if_highmem, Config0, false)
+                  andalso load_ctl:is_high_mem()
```

Comment:
do we need a `try` catch protection for this call during hot-upgrade ?

Reply:
for `load_ctl:is_high_mem()` ?

 I could make sure application `lc` gets loaded before loading this module right?